sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -86,6 +89,43 @@ class CollationStringExpressionsSuite extends QueryTest
     testRepeat("UNICODE_CI", 3, "abc", 2)
   }
 
+  test("substring check output type on explicitly collated string") {
```

Comment:
try covering some more edge cases in these tests, such as: empty strings, uppercase and lowercase mix, different byte-length chars, etc.

good example would be this PR: https://github.com/apache/spark/pull/45749

Reply:
@uros-db Please see what I have since added and let me know whether what to keep going, or whether we have enough!

Reply:
I think this will do it, thanks! Now just move over all tests to `CollationStringExpressionsSuite`, re-run any failing CI checks, and modify the PR title so we can open this up for review to Spark committers

Reply:
Sorry @uros-db, I didn't realize you wanted me to add the additional edge cases to this suite specifically. Are you sure you want me to move _all_ the new tests to `CollationStringExpressionsSuite`?

Reply:
yeah, let's put all these tests that are specific to string functions into the separate suite `CollationStringExpressionsSuite`

this way, `CollationSuite` won't get cluttered as we continue adding more and more support for various string expressions